Intersect.Server/Entities/Entity.cs

```diff
@@ -576,28 +590,29 @@ public virtual int CanMove(Direction moveDir)
                         {
                             if (en != null && en.X == tileX && en.Y == tileY && en.Z == Z && !en.Passable)
                             {
-                                return (int)EntityType.Event;
+                                blockerType = MovementBlockerType.Entity;
+                                entityType = EntityType.Event;
+                                return false;
                             }
                         }
                     }
                 }
             }
 
-            return IsTileWalkable(tile.GetMap(), tile.GetX(), tile.GetY(), Z);
-        }
-
-        protected virtual int IsTileWalkable(MapController map, int x, int y, int z)
-        {
-            //Out of bounds if no map
-            if (map == null)
+            if (IsTileWalkable(tile.GetMap(), tile.GetX(), tile.GetY(), Z))
             {
-                return -5;
+                blockerType = MovementBlockerType.NotBlocked;
+                entityType = EntityType.GlobalEntity;
+                return true;
             }
 
-            //Otherwise fine
-            return -1;
+            blockerType = MovementBlockerType.OutOfBounds;
+            entityType = EntityType.GlobalEntity;
+            return false;
         }
 
+        protected virtual bool IsTileWalkable(MapController map, int x, int y, int z) => map != null;
```

Comment:
This should return a `MovementBlockerType` and should be renamed to `GetTileBlocker`.

TBH I think the X/Y out of bounds check should probably happen here in the default implementation.